### PR TITLE
[MySQL Experience] (No.9) Fixed bug and code refactor

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/BasicLinkMySQLPanel.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        passwordSaveComboBox.setSize(lastColumnSize);
+        envPrefixTextField.setPreferredSize(lastColumnSize);
+        envPrefixTextField.setMaximumSize(lastColumnSize);
+        envPrefixTextField.setSize(lastColumnSize);
+        T config = supplier.get();
+        Optional.ofNullable(config.getService().getSubscription())
+                .ifPresent((subscription -> {
+                    this.subscriptionComboBox.setValue(new AzureComboBox.ItemReference<>(subscription.subscriptionId(), Subscription::subscriptionId));
+                    subscriptionComboBox.setForceDisable(true);
+                    subscriptionComboBox.setEditable(false);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/BasicLinkMySQLPanel.java:85]: May set editable(false) in `setForceDisable()`

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/BasicLinkMySQLPanel.java:85]: setForceDisable lies in super AzureComboBox, i think i should be better to let it pure and simple.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/BaseResourcePO.java [hunk 2]
```diff
@@ -13,8 +13,13 @@
 @Getter
 @AllArgsConstructor
 @EqualsAndHashCode
-public abstract class BaseServicePO {
+public abstract class BaseResourcePO {
 
     private String id;
+    private String resourceId;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/BaseResourcePO.java:19]: what is the difference from `id`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/BaseResourcePO.java [hunk 3]
```diff
@@ -13,8 +13,13 @@
...
     private ServiceType type;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/BaseResourcePO.java:20]: ResourceType?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/MySQLResourcePO.java [hunk 4]
```diff
@@ -7,27 +7,36 @@
...
+        this.database = JdbcUrl.from(url).getDatabase();
         this.passwordSave = passwordSave;
     }
 
+    @Override
+    public String getBusinessUniqueKey() {
+        return super.getBusinessUniqueKey() + "#" + database;
+    }
+
     public static class Builder {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/po/MySQLResourcePO.java:37]: use `@lombok.Buidler`

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**